docs/api-reference/next/server.md

```diff
@@ -88,6 +88,23 @@ The introduction of the `307` status code means that the request method is prese
 
 The `redirect()` method uses a `307` by default, instead of a `302` temporary redirect, meaning your requests will _always_ be preserved as `POST` requests.
 
+### Accessing environment variables
+
+`process.env` can be used to access [environment variables](docs/basic-features/environment-variables.md) from Middleware. However, these are evaluated at build time so only environment variables _actually_ used will be included.
+
+Any variables in `process.env` must be accessed directly, and **cannot** be destructured:
+
+```ts
+// This will not work
+const { NODE_ENV } = process.env
+// NODE_ENV is `undefined`
+console.log(NODE_ENV)
+// process.env is `{}`
+console.log(process.env)
+// Accessed directly, and not destructured works. process.env.NODE_ENV is `"development"` or `"production"`
```

Comment:
Should we start by showing what works?

Reply:
```
// This works when accessed directly. process.env.NODE_ENV is `"development"` or `"production"`
```

Reply:
That's a good idea. I like that way of thinking 🙌